### PR TITLE
Implement `lance.write_table` API and test the simplest round trips

### LINKED ISSUE #3: Provide WriteTable API in Python.
# Problem

Currently, the python package can read an `lance` file. We need to add write support in Python as well.

The `WriteTable` functionality has already been implemented in [C++ side](https://github.com/eto-ai/lance/blob/9db76594fac41e8150f03c100ecf315be5af93d1/cpp/include/lance/arrow/writer.h#L29). 

# Desired behavior

Implement `pylance.write_table(...)` API to wrap C++ `WriteTable` API, takes in an Pyarrow Table, and write to disk.

### PR DESCRIPTION
**Pull request by eddyxu**
* Added `lance.write_table()` API
* Test a round trip of writing data into `lance` data and read it back

Closes #3

### CODE REVIEW

File: python/lance/__init__.py [hunk 1]
```diff
@@ -13,3 +33,18 @@ def dataset(uri: str):
...
+
+
+def write_table(table: pa.Table, destination: Union[str, Path], primary_key: str):
+    """Write an Arrow Table into the destination.
+
+    Parameters
+    ----------
+    table : pa.Table
+        Apache Arrow Table
+    sink : str or `Path`
```
Comment (changhiskhan) [inline python/lance/__init__.py:45]: Match signature?

File: python/lance/__init__.py [hunk 2]
```diff
@@ -13,3 +33,18 @@ def dataset(uri: str):
     """
     fmt = LanceFileFormat()
     return ds.dataset(uri, format=fmt)
+
+
+def write_table(table: pa.Table, destination: Union[str, Path], primary_key: str):
```
Comment (changhiskhan) [inline python/lance/__init__.py:38]: maybe add a convenience to auto generate a pk column?

Comment (changhiskhan) [inline python/lance/__init__.py:38]: So this requires holding everything in memory first right? If we have a bunch of images on S3, does this mean we need to hold them all in Arrow memory to convert to lance format?

Comment (eddyxu) [reply python/lance/__init__.py:38]: Right, so there will be a `StreamWriter` which basically opens a `DatasetWriter` and write batch records one by one.

It is another set of interfaces tho.

Similar to parquet https://arrow.apache.org/docs/cpp/parquet.html#writing-parquet-files

Comment (eddyxu) [reply python/lance/__init__.py:38]: should we push that to the application / db level?

File: python/lance/__init__.py (hunk 1 above)
Comment (eddyxu) [reply python/lance/__init__.py:45]: Fixed

File: python/lance/__init__.py (hunk 2 above)
Comment (changhiskhan) [reply python/lance/__init__.py:38]: If we want people to use it as a python library then it's probably a good idea to have it. Could be in a wrapper function or something? Should also check for uniqueness there as well.

**Pull request merged**